### DISCUSSION TITLE
How to parse Connection String?

### DISCUSSION THREAD

**Discussion by skfd** [microsoft/PowerPlatform-DataverseServiceClient#200] [Q&A]
We used to be able to do `new CrmServiceClient(CONNECTION_STRING).OrganizationServiceProxy.ClientCredentials`. 

Now it seems we could  maybe do `DataverseConnectionStringProcessor.Parse(CONNECTION_STRING)`, but this static method is internal.

How do I parse connection string then?

Thank you!

**Comment by MattB-msft** [comment]
Out of curiosity,... why do you need to parse it?
You passed in the client credentials when you passed the connection string and would have had them at the time of creating the client.

**Comment by skfd** [comment]
> why do you need to parse it?

Porting an app and it has some credential tokenization logic that rewrites them.

> You passed in the client credentials

I did, but it's in a form of connection string, I can parse it by hand, but there's a piece of code that does that already )

**Reply by MattB-msft** [reply]
Ok, I generally understand why your asking for it.  We closed access to that down as it was an unnecessary exposer in our system.  Increasing our API surface to test and work with. 

For your use case, in the situation where there is a UID / PW in the connection string, your going to need to parse it on your side at this point.  We do not intended to re-expose that part of the internal api at this time.

**Reply by skfd** [reply]
that's no problem, just good to know, thank you!
